src/SimplifyCorrelatedDifferences.cpp

```diff
@@ -109,17 +109,6 @@ class SimplifyCorrelatedDifferences : public IRMutator {
         return visit_let<LetStmt, Stmt>(op);
     }
 
-    Stmt visit(const Store *op) override {
```

Comment:
Maybe worth capturing this PR's rationale in a comment here, so no tries to restore this logic in the future.

Reply:
I improved and expanded the top-level comment on the pass.